src/odemis/driver/emitter.py

```diff
@@ -57,14 +58,42 @@ def __init__(self, name, role, dependencies, **kwargs):
             self._child_idx[child] = len(spectra)
             spectra.extend(child.spectra.value)
             max_power.extend(child.power.range[1])
+            # Subscribe to each child power to update self.power
+            child.power.subscribe(self._updateMultiplexPower)
 
         # Child with the maximum power range
         self.power = model.ListContinuous(value=[0.0] * len(max_power),
-                                          range=(tuple([0.0] * len(max_power)), tuple(max_power),), unit="W")
-
+                                          range=(tuple([0.0] * len(max_power)), tuple(max_power),),
+                                          unit="W", cls=(int, long, float),)
+        self.power.subscribe(self._updateChildPower)
+        self._updateMultiplexPower(None)
         # info on which source is which wavelength
         self.spectra = model.ListVA(spectra, unit="m", readonly=True)
 
+    def _updateChildPower(self, power):
+        """
+        Whenever the self.power changes, the .power of the correct dependency is adjusted accordingly
+        """
+        # Prevent re-updating child power if power values are the same
+        # To avoid infinite loop from alternating subscriber calls
+        child_power = []
+        for child in self._child_idx.keys():
+            child_power.extend(child.power.value)
+        if power == child_power:
+            return
+        # Update child powers with new values
+        power = copy.deepcopy(power)  # Not to be attached to self.power.value
+        for child, idx in self._child_idx.items():
+            child.power.value[:] = power[idx:idx+len(child.power.value)]
```

Comment:
Please drop the `[:]`. `NotifyingList` is actually clever enough to detect it too, but it's easier to just set it explicitly.